backend/quatrix/quatrix.go

```diff
@@ -0,0 +1,1255 @@
+func (f *Fs) CreateDir(ctx context.Context, pathID, leaf string) (dirID string, err error) {
+	dir, err := f.createDir(ctx, pathID, leaf, false)
+	if err != nil {
+		return "", err
+	}
+
+	return dir.ID, nil
+}
+
+// Name of the remote (as passed into NewFs)
+func (f *Fs) Name() string {
+	return f.name
+}
+
+// Root of the remote (as passed into NewFs)
+func (f *Fs) Root() string {
+	return f.root
+}
+
+// String converts this Fs to a string
+func (f *Fs) String() string {
+	return f.description
+}
+
+// Precision return the precision of this Fs
+func (f *Fs) Precision() time.Duration {
+	return time.Microsecond
+}
+
+// Hashes returns the supported hash sets.
+func (f *Fs) Hashes() hash.Set {
+	return 0
+}
+
+// Features returns the optional features of this Fs
+func (f *Fs) Features() *fs.Features {
+	return f.features
+}
+
+// List the objects and directories in dir into entries.  The
+// entries can be returned in any order but should be for a
+// complete directory.
+//
+// dir should be "" to list the root, and should not have
+// trailing slashes.
+//
+// This should return ErrDirNotFound if the directory isn't
+// found.
+func (f *Fs) List(ctx context.Context, dir string) (entries fs.DirEntries, err error) {
+	directoryID, err := f.dirCache.FindDir(ctx, dir, false)
+	if err != nil {
+		return nil, err
+	}
+
+	folder, err := f.metadata(ctx, directoryID, true)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range folder.Content {
+		remote := path.Join(dir, f.opt.Enc.ToStandardName(file.Name))
+		if file.IsDir() {
+			f.dirCache.Put(remote, file.ID)
+
+			d := fs.NewDir(remote, time.Time(file.Modified)).SetID(file.ID).SetItems(file.Size)
+			// FIXME more info from dir?
+			entries = append(entries, d)
+		} else {
+			o := &Object{
+				fs:     f,
+				remote: remote,
+			}
+
+			err = o.setMetaData(&file)
+			if err != nil {
+				fs.Debugf(file, "failed to set object metadata: %s", err)
+			}
+
+			entries = append(entries, o)
+		}
+	}
+
+	return entries, nil
+}
+
+// NewObject finds the Object at remote.  If it can't be found
+// it returns the error fs.ErrorObjectNotFound.
+func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
+	return f.newObjectWithInfo(ctx, remote, nil)
+}
+
+// Creates from the parameters passed in a half finished Object which
+// must have setMetaData called on it
+//
+// # Returns the object, leaf, directoryID and error
+//
+// Used to create new objects
+func (f *Fs) createObject(ctx context.Context, remote string) (o *Object, leaf string, directoryID string, err error) {
+	// Create the directory for the object if it doesn't exist
+	leaf, directoryID, err = f.dirCache.FindPath(ctx, remote, true)
+	if err != nil {
+		return
+	}
+	// Temporary Object under construction
+	o = &Object{
+		fs:     f,
+		remote: remote,
+	}
+	return o, leaf, directoryID, nil
+}
+
+// Put the object into the container
+//
+// # Copy the reader in to the new object which is returned
```

Comment:
To avoid `go fmt` putting this `#` in, make sure the previous sentence has a full stop! Or is it this one? Anyway try adding full stops and you can remove the `#`.